pkg/sensors/exec/cgroups_test.go

```diff
@@ -323,3 +502,119 @@ func TestCgroupEventMkdirRmdir(t *testing.T) {
 	_, err = cgrouptrackmap.LookupTrackingCgroup(cgrpMapPath, cgrpTrackingId)
 	assert.Error(t, err)
 }
+
+func testCgroupv2HierarchyInUnified(ctx context.Context, t *testing.T,
+	cgroupRoot string, cgroupHierarchy []cgroupHierarchy, trackingCgrpLevel uint32,
+	triggers []func()) {
+
+	t.Logf("Test %s running in %s", t.Name(), cgroups.CgroupModeCode(cgroups.CGROUP_UNIFIED).String())
+	t.Logf("Test %s cgroup mount point: %s -> %s", t.Name(), cgroups.GetCgroupFSPath(), cgroupRoot)
+
+	// Ensure that we do not have a test cgroup hierarchy since cgroupfs is shared
+	sharedCgroupPath := filepath.Join(cgroups.GetCgroupFSPath(), cgroupHierarchy[0].path)
+	t.Logf("Test %s cleaning up cgroup: %s", t.Name(), sharedCgroupPath)
+	os.RemoveAll(sharedCgroupPath)
+	_, err := os.Stat(sharedCgroupPath)
+	require.Error(t, err)
+
+	t.Cleanup(func() {
+		cgroupRmdir(t, cgroupRoot, "", cgroupHierarchy[0].path)
```

Comment:
Yes it is created part of the `triggerCgroupMkdir` trigger when we iterate over the cgroupHierarchy, so this is an extra cleanup in case the the `triggerCgroupRmdir` fails or exits before which has a different purpose anyway, fine to have it here.

Reply:
Why don't we add the cleanup where it is created? If it fails, t.Cleanup() will run.

Reply:
Cause we have tests in the middle, cleanup will generate cgroup events, we don't want that !

> triggerCgroupRmdir fails or exits before which has a different purpose anyway

The purpose of those is cgroup events.

The purpose of the cleanup, is clean up things, where we should have terminated processing cgroup events.